src/Transport/ClosableTransportInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Transport;
+
+use GuzzleHttp\Promise\PromiseInterface;
+
+/**
+ * All classes implementing this interface will support sending events on-demand.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+interface ClosableTransportInterface
```

Comment:
Why is this not implemented in the Client?
It should basically call `flush` internally with the exception that the client disables itself after the close call.

Reply:
Implementing in the transport it's needed to mark that the transport exposes the `close` call. The client exposes the `flush` method with `FlushableClientInterface`.

Reply:
Exactly, I did not implement the `close` method on the client but the `flush` method (difference is that the latter does not dispose the client). Of course the transport must have a method to drain the internal queue of events on demand and since I could not add the method on the existing interface due to this action being a BC I had to create a new one that we can drop in `3.0` leaving only the former. I'm not sure it makes sense to support the `close` method